### PR TITLE
Add NoWASM label

### PR DESCRIPTION
**Pull request by DrewScoggins**

### CODE REVIEW

File: src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs [hunk 1]
```diff
@@ -66,6 +66,7 @@ public int Read()
         }
         
         [Benchmark]
+        [BenchmarkCategory(Categories.NoWASM)]
```
Comment (adamsitnik) [inline src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs:69]: are file operations not supported in general on WASM or just the async ones?

File: src/benchmarks/micro/libraries/System.Threading/Perf.CancellationToken.cs [hunk 2]
```diff
@@ -17,6 +17,7 @@ public class Perf_CancellationToken
         public void RegisterAndUnregister_Serial() => _token.Register(() => { }).Dispose();
 
         [Benchmark(OperationsPerInvoke = 1_000_000)]
+        [BenchmarkCategory(Categories.NoWASM)]
```
Comment (adamsitnik) [inline src/benchmarks/micro/libraries/System.Threading/Perf.CancellationToken.cs:20]: For the multi-threaded tests, we could introduce a `multithreaded` category and use it to exclude the benchmarks when affinitizing to a single core. Sth like:

```cmd
--filter * --category-exclusion-filter multithreaded --affinity X
```

And the full WASM filter would be:

```cmd
--filter * --category-exclusion-filter nowasm multithreaded
```

**Comment by adamsitnik** [comment]
btw @buyaa-n is currently working on an analyzer that is going to print warnings for os|platform specific APIs. When it's ready we could use it here to verify if our "nowasm" category is complete

File: src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs (hunk 1 above)
Comment (DrewScoggins) [reply src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs:69]: File operations do not seem to be allowed on WASM. From what I was able to glean from reading you need to hit a web endpoint if you want to access files.

Comment (DrewScoggins) [reply src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs:69]: So maybe that would be something we could do long term, but for now we can just turn them off.

**Pull request merged**